server/plugin.go

```diff
@@ -24,13 +25,17 @@ var SentryDSN string
 
 type Plugin struct {
 	plugin.MattermostPlugin
-
+	BotUserID string
 	handler http.Handler
 	running bool
 }
 
 func (p *Plugin) OnActivate() error {
 	config.Mattermost = p.API
+	
+	if err := p.setupBot(); err !=nil {
```

Comment:
Can we call this function from `OnConfigurationChange()` instead? This will remove the need of `Plugin.BotUserID` variable.